code/datums/traits/_trait.dm

```diff
@@ -0,0 +1,82 @@
+//every trait in this folder should be coded around being applied on spawn
+/datum/trait
+	var/name = "Test Trait"
+	var/desc = "This is a test trait."
+	var/value = 0
+	var/human_only = TRUE
+	var/gain_text
+	var/lose_text
+	var/mob_trait //if applicable, apply and remove this "mob trait", i.e. GOTTAGOFAST
+	var/mob/living/trait_holder
+	var/list/conflicts_with = list()
+
+/datum/trait/New(mob/living/trait_mob)
+	..()
+	if(!trait_mob || (human_only && !ishuman(trait_mob)) || trait_mob.has_trait(type))
+		qdel(src)
+	trait_holder = trait_mob
+	SStraits.trait_objects += src
+	to_chat(trait_holder, gain_text)
+	trait_holder.roundstart_traits += src
+	if(mob_trait)
+		trait_holder.add_trait(mob_trait, ROUNDSTART_TRAIT)
+	START_PROCESSING(SStraits, src)
+	add()
+
+/datum/trait/Destroy()
+	STOP_PROCESSING(SStraits, src)
+	to_chat(trait_holder, lose_text)
+	remove()
+	trait_holder.roundstart_traits -= src
+	if(mob_trait)
+		trait_holder.remove_trait(mob_trait, ROUNDSTART_TRAIT, TRUE)
+	SStraits.trait_objects -= src
+	return ..()
+
+/datum/trait/proc/add() //special "on add" effects
+/datum/trait/proc/remove() //special "on remove" effects
+/datum/trait/proc/on_process() //process() has some special checks, so this is the actual process
+
+/datum/trait/process()
+	if(!trait_holder)
+		qdel(src)
+	on_process()
+
+/*
+
+Commented version of Nearsighted to help you add your own traits
+Use this as a guideline
+
+/datum/trait/nearsighted
```

Comment:
neersighted

Reply:
is this a joke

Reply:
is this loss?

Reply:
this should be called trait/errorage in honour of the time he tried to make all engineers shortsighted lol

Reply:
It was only a 25% chance, thank you very MUCH.